noggin/representation/user.py

```diff
@@ -67,3 +67,24 @@ def rhbz_mail(self):
     @property
     def last_password_change(self):
         return self._attr('krblastpwdchange')
+
+    def get_properties(self):
+        """
+        Returns a dict of all the properties defined above and their values
+        """
+        return {
+            key: getattr(self, key)
+            for key, value in self.__class__.__dict__.items()
+            if isinstance(value, property)
+        }
+
+    def diff_against(self, other):
+        """
+        Compares two instances of the User class, and returns the properties
+        with values that are different
+        """
```

Comment:
You could move that method to the parent `Representation` class, so it's available to all our entity representations.
Also, please add a check on the nature of the object that we're diffing with, for example:
```python
if not isinstance(other, self.__class__):
    raise ValueError(f"Can't diff a {self.__class__.__name__} instance against a {other.__class__.__name__} instance")
```

Reply:
Done, and re-formatted the tests a bit so i could test this.